data-prepper-api/src/main/java/com/amazon/dataprepper/model/buffer/Buffer.java

```diff
@@ -32,6 +32,14 @@
      */
     void write(T record, int timeoutInMillis) throws TimeoutException;
 
+    /**
+     * Atomically writes collection of records into the buffer
```

Comment:
Should we state "atomically" in the Javadocs? Future implementations of Buffer may have difficulty with this requirement. Perhaps the requirement should be: "If this method successfully returns, all items were written. Some implementations may write some items and still fail. Implementors should make this atomic if possible."

Reply:
The purpose of introducing this new API is to do atomic batch writing. Non-atomic writing can be achieved by calling existing `write` method repeatedly. At the source implementation level, once could choose to return partial success response or not depending on whether write or writeAll is invoked. Another issue of allowing partial success with this API is one cannot tell which data is successfully written and thus making this API valueless compared to `write`.

Therefore, I tend to keep the description as it is until future implementations are found to be accommodated.

Reply:
Sounds good. This works for me.